infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -69,70 +75,73 @@
 
     final ExecutorService executor;
 
+    final WorkflowFactory workflowFactory;
+
     public Orchestrator(@Nonnull SingletonResource<CorfuRuntime> runtime,
-                        @Nonnull ServerContext serverContext) {
+                        @Nonnull ServerContext serverContext,
+                        @Nonnull WorkflowFactory workflowFactory) {
         this.serverContext = serverContext;
-        this.getRuntime = runtime;
-
-        executor = Executors.newFixedThreadPool(Runtime.getRuntime()
-                .availableProcessors(), new ThreadFactory() {
-
-            final AtomicInteger threadNumber = new AtomicInteger(0);
-
-            @Override
-            public Thread newThread(Runnable r) {
-                Thread thread = new Thread(r);
-                thread.setDaemon(true);
-                String threadName = serverContext.getThreadPrefix() + "orchestrator-"
-                        + threadNumber.getAndIncrement();
-                thread.setName(threadName);
-                thread.setUncaughtExceptionHandler(this::handleUncaughtException);
-                return thread;
-            }
+        this.workflowFactory = workflowFactory;
+        getRuntime = runtime;
 
-            void handleUncaughtException(Thread t, @Nonnull Throwable e) {
-                log.error("handleUncaughtException[{}]: Uncaught {}:{}",
-                        t.getName(),
-                        e.getClass().getSimpleName(),
-                        e.getMessage(),
-                        e);
-            }
-        });
-    }
+        executor = serverContext.getExecutorService(Runtime.getRuntime().availableProcessors(),
+                new ThreadFactory() {
+
+                    final AtomicInteger threadNumber = new AtomicInteger(0);
 
-    public void handle(@Nonnull CorfuPayloadMsg<OrchestratorMsg> msg,
-                       @Nonnull ChannelHandlerContext ctx,
-                       @Nonnull IServerRouter r) {
+                    @Override
+                    public Thread newThread(Runnable r) {
+                        Thread thread = new Thread(r);
+                        thread.setDaemon(true);
+                        String threadName = serverContext.getThreadPrefix() + "orchestrator-"
+                                + threadNumber.getAndIncrement();
+                        thread.setName(threadName);
+                        thread.setUncaughtExceptionHandler(this::handleUncaughtException);
+                        return thread;
+                    }
+
+                    void handleUncaughtException(Thread t, @Nonnull Throwable e) {
```

Comment:
This can be replaced with the getExecutor helper methods in server context, right?

Reply:
Can you clarify? The `getExecutor` helper method is being called on line 87.